runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -568,11 +568,19 @@ ClassFileOracle::walkRecordComponents(J9CfrAttributeRecord *attrib)
 				break;
 			}
 			case CFR_ATTRIBUTE_RuntimeVisibleAnnotations: {
-				_recordComponentsInfo[i].annotationsAttribute = (J9CfrAttributeRuntimeVisibleAnnotations *) recordComponentAttr;
+				J9CfrAttributeRuntimeVisibleAnnotations *recordComponentAnnotations = (J9CfrAttributeRuntimeVisibleAnnotations *)recordComponentAttr;
+				if (0 == recordComponentAnnotations->rawDataLength) {
+					walkAnnotations(recordComponentAnnotations->numberOfAnnotations, recordComponentAnnotations->annotations, 0);
+				}
+				_recordComponentsInfo[i].annotationsAttribute = recordComponentAnnotations;
 				break;
 			}
 			case CFR_ATTRIBUTE_RuntimeVisibleTypeAnnotations: {
-				_recordComponentsInfo[i].typeAnnotationsAttribute = (J9CfrAttributeRuntimeVisibleTypeAnnotations *) recordComponentAttr;
+				J9CfrAttributeRuntimeVisibleTypeAnnotations *recordComponentTypeAnnotations = (J9CfrAttributeRuntimeVisibleTypeAnnotations *)recordComponentAttr;
+				if (0 == recordComponentTypeAnnotations->rawDataLength) {
+					walkTypeAnnotations(recordComponentTypeAnnotations->numberOfAnnotations, recordComponentTypeAnnotations->typeAnnotations);
```

Comment:
The knownAnnotations argument is for tracking specialAnnotations.  It shouldn't be set unless the code needs to do something with the "found" annotations - and this code isn't as it isn't checking the return value
```suggestion
					walkTypeAnnotations(recordComponentTypeAnnotations->numberOfAnnotations, 0);
```

Reply:
It should be correct. Only `walkAnnotations` has the `knownAnnotations` parameter.

Reply:
Sorry, my bad.  I misread the code